src/hapi/pipelines/utilities/operational_presence.py

```diff
@@ -0,0 +1,201 @@
+        hapi_metadata = self.get_hapi_metadata()
+        resource_ref = hapi_metadata["hdx_id"]
+        reference_period_start = hapi_metadata["reference_period"]["startdate"]
+        reference_period_end = hapi_metadata["reference_period"]["enddate"]
+        for row in self.data:
+            admin_code = row.get("#adm2+code")
+            admin_level = "2"
+            if not admin_code:
+                admin_code = row.get("#adm1+code")
+                admin_level = "1"
+            org_name = row.get("#org+name")
+            org_acronym = row.get("#org+acronym")
+            orgtype_name = row.get("#org+type")
+            sector_info = row.get("#sector")
+            orgtype_code = _get_orgtype_code(orgtype_name, orgtype.data)
+            if orgtype_code == "":
+                # TODO: What do we do if the org type is not in the orgtype table?
+                logger.error(f"Org type {orgtype_name} not in table")
+            # TODO: find out how unique orgs are. Currently checking that combo of acronym/name/type is unique
+            if (
+                    org_acronym is not None
+                    and org_name is not None
+                    and (org_acronym, org_name, orgtype_code) not in org.data
+            ):
+                org.populate_single(
+                    acronym=org_acronym,
+                    orgname=org_name,
+                    orgtype=orgtype_code,
+                    reference_period_start=reference_period_start,
+                    reference_period_end=reference_period_end,
+                )
+
+            sector_name, sector_code = _get_sector_info(sector_info, sector.data)
+            if sector_code == "" or sector_name == "":
+                # TODO: What do we do if the sector is not in the sector table?
+                logger.error(f"Sector {sector_info} not in table")
+
+            if admin_level == "1":
+                admin2_code = get_admin2_to_admin1_connector_code(
+                    admin1_code=admin_code
+                )
+            elif admin_level == "2":
+                admin2_code = admin_code
+            operational_presence_row = DBOperationalPresence(
+                resource_ref=resource_ref,
+                org_ref=org.data[(org_acronym, org_name, orgtype_code)],
+                sector_code=sector_code,
+                admin2_ref=admins.data[admin2_code],
+                reference_period_start=reference_period_start,
+                reference_period_end=reference_period_end,
+                # TODO: For v2+, add to scraper
+                source_data="not yet implemented",
+            )
+            self._session.add(operational_presence_row)
+        self._session.commit()
+
+
+def _get_orgtype_code(orgtype: str, orgtype_data: Dict) -> str:
+    # TODO: implement fuzzy matching of org types
+    orgtype_code = orgtype_data.get(orgtype, "")
+    return orgtype_code
+
+
+def _get_sector_info(sector_info: str, sector_data: Dict) -> (str, str):
```

Comment:
See my comments in sector.py about the lookup table - I think this can be highly simplified